linkerd/app/outbound/src/ingress.rs

```diff
@@ -79,142 +88,143 @@ impl<H> Outbound<H> {
-                .push(svc::FailFast::layer("HTTP Server", dispatch_timeout))
-                .push(rt.metrics.http_errors.clone())
-                .push(errors::layer())
-                .push(http_tracing::server(rt.span_sink, trace_labels()))
-                .push(http::BoxResponse::layer()),
-        )
-        .instrument(|a: &http::Accept| debug_span!("http", v = %a.protocol))
-        .push(http::NewServeHttp::layer(h2_settings, rt.drain))
-        .push_request_filter(|(http, accept): (Option<http::Version>, _)| {
-            http.map(|h| http::Accept::from((h, accept)))
-                .ok_or(IngressHttpOnly)
-        })
-        .push_cache(cache_max_idle_age)
-        .push_map_target(detect::allow_timeout)
-        .push(svc::BoxNewService::layer())
-        .push(detect::NewDetectService::layer(
-            detect_protocol_timeout,
-            http::DetectHttp::default(),
-        ))
-        .push(rt.metrics.transport.layer_accept())
-        .instrument(|a: &tcp::Accept| info_span!("ingress", orig_dst = %a.orig_dst))
-        .push_map_target(|a: T| {
-            let orig_dst = Param::<OrigDstAddr>::param(&a);
-            tcp::Accept::from(orig_dst)
-        })
-        .push_on_response(svc::BoxService::layer())
-        .push(svc::BoxNewService::layer())
-        .check_new_service::<T, I>()
-        .into_inner()
-    }
-}
-
-// === impl Target ===
 
-impl TryFrom<(Option<profiles::Receiver>, Target)> for http::Logical {
-    type Error = ProfileRequired;
-    fn try_from(
-        (profile, Target { version, .. }): (Option<profiles::Receiver>, Target),
-    ) -> Result<Self, Self::Error> {
-        let profile = profile.ok_or(ProfileRequired)?;
-        let logical_addr = profile.borrow().addr.clone().ok_or(ProfileRequired)?;
-
-        Ok(Self {
-            profile,
-            protocol: version,
-            logical_addr,
-        })
-    }
-}
-
-impl From<(tls::NoClientTls, Target)> for http::Endpoint {
-    fn from((reason, Target { dst, version }): (tls::NoClientTls, Target)) -> Self {
-        // XXX This is a hack to fix caching when an dst-override is set.
-        let addr = Remote(ServerAddr(([0, 0, 0, 0], dst.port()).into()));
-
-        Self {
-            addr,
-            metadata: Metadata::default(),
-            tls: Conditional::None(reason),
-            logical_addr: None,
-            protocol: version,
-            // The endpoint is HTTP and definitely not opaque.
-            opaque_protocol: false,
-        }
-    }
-}
-
-// === TargetPerRequest ===
-
-impl TargetPerRequest {
-    fn accept(a: http::Accept) -> Self {
-        Self(a)
-    }
-}
-
-impl<B> svc::stack::RecognizeRoute<http::Request<B>> for TargetPerRequest {
-    type Key = Target;
+        // Route requests with destinations that can be discovered via the `l5d-dst-override` header
+        // through a logical stack and route requests without the `l5d-dst-override` header through
```

Comment:
unimportant nit
```suggestion
        // through a logical stack, and routes requests without the `l5d-dst-override` header through
```

Reply:
that conjugation doesn't work here. In any case, updated the comment.